### PR TITLE
Updating mounting doc

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
check here for [view](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b4d9297e8beb9fc62867c7678763ea846ab5c75/docs/mounting.md)

### Testing
N/A

### CODE REVIEW

File: docs/mounting.md [hunk 1]
```diff
@@ -1,179 +1,202 @@
...
-`storage-full` access scope. (See [here][gce-service-accounts] for details on
-VM service accounts.) When gcsfuse is run from such a VM, it automatically has
-access to buckets owned by the same project as the VM.
+        gcloud auth application-default login
+        gcloud auth login
+        gcloud auth list
 
-When testing, especially on a developer machine, credentials can also be
-configured using the [gcloud tool][]:
+   Alternatively, you can authenticate Cloud Storage FUSE by setting the --key-file flag to the path of a JSON key file, which you can download from the Google Cloud console. You can also set the GOOGLE_APPLICATION_CREDENTIALS environment variable to the path of the JSON key.
```
Comment (sethiay) [inline docs/mounting.md:10]: nit: Even though not in user guide, you can put --key-file and GOOGLE_APPLICATION_CREDENTIALS in code block

**Review by sethiay — APPROVED** [review]

File: docs/mounting.md [hunk 2]
```diff
@@ -1,179 +1,202 @@
...
+    
+    gcsfuse my-bucket /usr/me/myself
 
-## Mounting
+# Static Mounting
 
-There are two ways to mount gcsfuse. If you only use one known bucket, you may
-use static mounting. If you want to mount all the accessible buckets, you may
-use dynamic mounting.
+Static mounting means mounting a specific bucket. For example, say I want to mount the bucket “my-bucket” to the directory /usr/me/myself
```
Comment (sethiay) [inline docs/mounting.md:31]: please replace /usr/me/myself to /path/to/mount/point at all places. (the user guide is also updated). Sorry to tell now but please also look in the docs published and correct there as well.

File: docs/mounting.md (hunk 1 above)
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md [hunk 3]
```diff
@@ -1,179 +1,202 @@
...
+The base syntax for using Cloud Storage FUSE is:
 
-[gce]: https://cloud.google.com/compute/
-[gce-service-accounts]: https://cloud.google.com/compute/docs/authentication
-[gcloud tool]: https://cloud.google.com/sdk/gcloud/
-[app-default-credentials]: https://developers.google.com/identity/protocols/application-default-credentials#howtheywork
+    gcsfuse [global options] [bucket] mountpoint
 
-# Basic usage
+Where [global options] are optional specific flags you can pass (use gcsfuse --help), [bucket] is the optional name of your bucket, and ‘mountpoint’ is the directory on your machine that you are mounting the bucket to. For example:
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md [hunk 4]
```diff
@@ -1,179 +1,202 @@
...
+    gcsfuse -o ro my-bucket /usr/me/myself
 
-### Foreground
+# Foreground
 
-After the gcsfuse tool exits, you should be able to see your bucket contents if
-you run `ls /path/to/mount/point`. If you would prefer the tool to stay in the
-foreground (for example to see debug logging), run it with the `--foreground`
-flag.
+After Cloud Storage FUSE exits, you should be able to see your bucket contents if you run ls /path/to/mount/point. If you would prefer the tool to stay in the foreground (for example to see debug logging), run it with the --foreground flag.
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md [hunk 5]
```diff
@@ -1,179 +1,202 @@
...
+| -stat-cache-capacity value      | How many entries can the stat cache hold (impacts memory consumption) (default: 4096)|
+| --stat-cache-ttl value          |How long to cache StatObject results and inode attributes. (default: 1m0s)|
+| --type-cache-ttl value          |How long to cache name -> file/dir mappings in directory inodes. (default: 1m0s)|
+| --max-retry-duration value      |The operation will be retried till the value of max-retry-duration. (default: 30s)|
+|--retry-multiplier value|Param for exponential backoff algorithm, which is used to increase waiting time b/w two consecutive retries. (default: 2)|
+|--experimental-local-file-cache|Experimental: Cache GCS files on local disk for reads.|
+|--temp-dir value|Path to the temporary directory where writes are staged prior to upload to Cloud Storage. (default: system default, likely /tmp)|
+|--client-protocol value|The protocol used for communicating with the GCS backend. Value can be 'http1' (HTTP/1.1) or 'http2' (HTTP/2).|
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md (hunk 2 above)
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md [hunk 6]
```diff
@@ -1,179 +1,202 @@
...
 
-    ls /path/to/mount/point/my-bucket/foo/
+# Mounting a specific directory in a Cloud Storage bucket instead of the entire bucket
 
-you will have `my-bucket` dynamically mounted.
+By default, Cloud Storage FUSE mounts the entire contents and directory structure within a bucket. To mount only a specific directory, pass the --only-dir option. For example, if ‘my-bucket’ contains the path ‘my-bucket/a/b’ to mount only a/b to my local directory /usr/me/myself:
 
-It is illegal to list subdirectory at the root level in the dynamic mounted file
-system, such as
+    gcsfuse --only-dir my-bucket a/b /usr/me/myself
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

File: docs/mounting.md [hunk 7]
```diff
@@ -1,179 +1,202 @@
...
+
+The Cloud Storage FUSE installation process installs a helper understood by the mount command to your system at the path /sbin/mount.gcsfuse, allowing you to mount your bucket using the mount command (on Linux, only root can do this). Example:
+
+    mount -t gcsfuse -o rw,user my-bucket /path/to/mount/point
+
+You can also add entries to your /etc/fstab file like the following:
+
+    my-bucket /mount/point gcsfuse rw,noauto,user
+
+Afterward, you can run mount /mount/point as a non-root user.
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply docs/mounting.md]: fixed

**Pull request merged**